node/src/components/consensus/era_supervisor.rs

```diff
@@ -881,60 +701,68 @@ where
             proposed_block,
             valid,
         } = resolve_validity;
-        self.era_supervisor.metrics.proposed_block();
+        self.metrics.proposed_block();
         let mut effects = Effects::new();
         if !valid {
             warn!(
                 %sender,
                 era = %era_id.value(),
                 "invalid consensus value; disconnecting from the sender"
             );
-            effects.extend(self.disconnect(sender));
+            effects.extend(self.disconnect(effect_builder, sender));
         }
         if self
-            .era_supervisor
             .active_eras
             .get_mut(&era_id)
             .map_or(false, |era| era.resolve_validity(&proposed_block, valid))
         {
-            effects.extend(self.delegate_to_era(era_id, |consensus| {
-                consensus.resolve_validity(proposed_block, valid, Timestamp::now())
-            }));
+            effects.extend(
+                self.delegate_to_era(effect_builder, rng, era_id, |consensus| {
+                    consensus.resolve_validity(proposed_block, valid, Timestamp::now())
+                }),
+            );
         }
         effects
     }
 
-    fn handle_consensus_outcomes<T>(&mut self, era_id: EraId, outcomes: T) -> Effects<Event<I>>
+    fn handle_consensus_outcomes<REv: ReactorEventT<I>, T>(
+        &mut self,
+        effect_builder: EffectBuilder<REv>,
+        rng: &mut NodeRng,
+        era_id: EraId,
+        outcomes: T,
+    ) -> Effects<Event<I>>
     where
         T: IntoIterator<Item = ProtocolOutcome<I, ClContext>>,
     {
         outcomes
             .into_iter()
-            .flat_map(|result| self.handle_consensus_outcome(era_id, result))
+            .flat_map(|result| self.handle_consensus_outcome(effect_builder, rng, era_id, result))
             .collect()
     }
 
     /// Returns `true` if any of the most recent eras has evidence against the validator with key
     /// `pub_key`.
     fn has_evidence(&self, era_id: EraId, pub_key: PublicKey) -> bool {
-        self.era_supervisor
-            .iter_past(era_id, self.era_supervisor.bonded_eras())
+        self.iter_past(era_id, 1)
```

Comment:
Can this `1` be extracted to a constant as well?

Reply:
Done in https://github.com/casper-network/casper-node/pull/2127/commits/f417ff94706385eda1d3d134cadcc7fec3f6dfd0.